src/state/slices/txHistorySlice/txHistorySlice.ts

```diff
@@ -301,15 +293,15 @@ export const txHistoryApi = createApi({
         const foxyArgs = { adapter, foxyAddresses, providerUrl }
         const foxyApi = new FoxyApi(foxyArgs)
 
-        foxyTokenContractAddressWithBalances.forEach(async tokenContractAddress => {
+        for (const tokenContractAddress of foxyTokenContractAddressWithBalances) {
```

Comment:
Promise.all to improve performance of fetching all of these possible rebase histories?

Reply:
Addressed in https://github.com/shapeshift/web/pull/1892/commits/effc6bd85de59333b65a003f04274994f77b8b5d